rpi/controller.py

```diff
@@ -52,43 +50,40 @@ def check_access(rules, time:datetime=None):
         # Access is never granted for that person
         reason = "access is never granted"
         result = False
-    else:
-        # Day and time rules are associated with that person
-        
-        # If a time isn't passed as argument, check against the current time
-        if time is None:
-            time=datetime.now()
-        
-        # Process rules
-        thisDay = time.weekday()
-        thisHour = time.hour
-        thisMinute = time.minute
-        
-        for rule in rules:
-            # Check day
-            if rule[0][0] == days[thisDay]:
-                # Day matches
-                if not rule[0][1]:
-                    # If times are not specified, access is granted for that whole day
-                    result = True
-                else:
-                    # If times are specified, check that we are within the interval
-                    startHour, startMinute = map(int, rule[0][1].split(':'))
-                    endHour, endMinute = map(int, rule[0][2].split(':'))
-                    
-                    if (startHour*60+startMinute <= thisHour*60+thisMinute <= endHour*60+endMinute):
-                        result = True
-        
-        if (result):
-            if (verbose_log):
-                reason = "%s (%s) is during allowed hours %s" % (time, days[thisDay], rules)
-            else:
-                reason = "it is during allowed hours"
-        else:
-            if (verbose_log):
-                reason = "%s (%s) is not during allowed hours %s" % (time, days[thisDay], rules)
+    
+    # leave this now, so we don't need an else spot.
+    if rules[0][0][0] in ['ALWAYS', 'NEVER']:
```

Comment:
Opinion: It would read a bit easier if each of the above if and elif statements just returned out the end.
We add 2 lines here to save repeating one line twice.